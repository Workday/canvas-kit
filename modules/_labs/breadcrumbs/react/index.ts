import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
