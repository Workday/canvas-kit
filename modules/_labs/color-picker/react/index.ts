export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
