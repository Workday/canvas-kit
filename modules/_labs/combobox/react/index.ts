import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
