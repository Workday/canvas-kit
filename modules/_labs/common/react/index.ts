export * from './lib/Flex';
export * from './lib/Box';
