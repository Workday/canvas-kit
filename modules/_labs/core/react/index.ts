import type from './lib/type';
import space from './lib/space';
export default type;
export {type, space};
export * from './lib/type';
export * from './lib/StaticStates';
