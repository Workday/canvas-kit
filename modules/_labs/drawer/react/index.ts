import Drawer from './lib/Drawer';
import DrawerHeader from './lib/DrawerHeader';
export {Drawer, DrawerHeader};
export default Drawer;
export * from './lib/Drawer';
