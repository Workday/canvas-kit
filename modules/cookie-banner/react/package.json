{
  "name": "@workday/canvas-kit-react-cookie-banner",
  "version": "3.9.1",
  "description": "Canvas styled cookie banner",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/cookie-banner/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "cookie-banner"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@emotion/core": "^10.0.28",
    "@emotion/styled": "^10.0.27",
    "@workday/canvas-kit-react-button": "^3.9.1",
    "@workday/canvas-kit-react-core": "^3.9.1"
  }
}
