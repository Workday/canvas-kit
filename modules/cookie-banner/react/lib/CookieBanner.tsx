/** @jsx jsx */
import {css, jsx} from '@emotion/core';
import * as React from 'react';
import styled from '@emotion/styled';
import {colors, commonColors, type, spacing} from '@workday/canvas-kit-react-core';
import Button from '@workday/canvas-kit-react-button';

export interface CookieBannerProps {
  /**
   * If true, set the CookieBanner to the closed state.
   * @default false
   */
  isClosed?: boolean;

  /**
   * The function called when cookies are accepted via the CookieBanner. This callback function should set `isClosed` to `true`.
   */
  onAccept: (e: React.MouseEvent<HTMLButtonElement>) => void;

  /**
   * The function called when the CookieBanner settings button is clicked. The settings button is automatically displayed if this callback function is provided.
   */
  onClickSettings?: (e: React.MouseEvent<HTMLButtonElement>) => void;

  /**
   * The notice text or element of the CookieBanner.
   * @default CookieBanner.DefaultNotice
   */
  notice: string | React.ReactNode;

  /**
   * The text of the CookieBanner settings button.
   * @default 'Cookie Settings'
   */
  settingsLabel: string;
}

const Banner = styled('div')(
  type.body2,
  {
    backgroundColor: commonColors.background,
    borderTop: `1px solid ${colors.soap400}`,
    display: 'flex',
    padding: spacing.m,
    alignItems: 'center',
    justifyContent: 'space-between',
    position: 'fixed',
    bottom: 0,
    left: 0,
    right: 0,
    zIndex: 99,
    transition: 'transform 0.2s ease-out',
    '@media (max-width: 450px)': {
      flexDirection: 'column',
      alignItems: 'stretch',
      textAlign: 'center',
      padding: `${spacing.s} 0`,
    },
  },
  ({isClosed}: Pick<CookieBannerProps, 'isClosed'>) =>
    isClosed ? {transform: 'translateY(100%)'} : null
);

const BannerItem = styled('div')({
  marginLeft: spacing.s,
  marginRight: spacing.s,
  '@media (max-width: 450px)': {
    '&:not(:first-child)': {
      marginTop: spacing.s,
    },
  },
});

const rowStyle = css({
  display: 'flex',
});

const CookieSettings = styled('button')(type.body2, type.link, {
  marginRight: spacing.s,
  border: 0,
  fontWeight: 500,
  whiteSpace: 'nowrap',
  cursor: 'pointer',
  padding: 0,
  height: '0%',
  alignSelf: 'center',
});

export default class CookieBanner extends React.Component<CookieBannerProps> {
  public static DefaultNotice =
    'We use cookies to ensure that we give you the best experience on our website. If you continue without changing your settings, weâ€™ll assume that you are willing to receive cookies.';

  public static defaultProps = {
    notice: CookieBanner.DefaultNotice,
    settingsLabel: 'Cookie Settings',
  };

  public render(): React.ReactNode {
    const {isClosed, onAccept, onClickSettings, notice, settingsLabel, ...elemProps} = this.props;

    return (
      <Banner isClosed={isClosed} {...elemProps}>
        <BannerItem>{notice}</BannerItem>
        <BannerItem css={rowStyle}>
          {onClickSettings && (
            <CookieSettings onClick={onClickSettings}>{settingsLabel}</CookieSettings>
          )}
          <Button onClick={onAccept} variant={Button.Variant.Primary} grow={true}>
            Continue
          </Button>
        </BannerItem>
      </Banner>
    );
  }
}
