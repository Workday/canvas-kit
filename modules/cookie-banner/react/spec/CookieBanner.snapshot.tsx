import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CookieBanner from '../lib/CookieBanner';

function noop() {
  // noop
}

describe('Cookie Banner Snapshots', () => {
  test('renders a CookieBanner', () => {
    const component = renderer.create(<CookieBanner onAccept={noop} />);
    expect(component).toMatchSnapshot();
  });

  test('renders a closed CookieBanner', () => {
    const component = renderer.create(<CookieBanner onAccept={noop} isClosed={true} />);
    expect(component).toMatchSnapshot();
  });

  test('renders a CookieBanner with settings', () => {
    const component = renderer.create(<CookieBanner onAccept={noop} onClickSettings={noop} />);
    expect(component).toMatchSnapshot();
  });

  test('renders a closed CookieBanner with settings', () => {
    const component = renderer.create(
      <CookieBanner onAccept={noop} onClickSettings={noop} isClosed={true} />
    );
    expect(component).toMatchSnapshot();
  });
});
