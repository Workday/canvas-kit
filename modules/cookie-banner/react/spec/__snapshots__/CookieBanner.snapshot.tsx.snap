// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Banner Snapshots renders a CookieBanner 1`] = `
.emotion-8 {
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffffff;
  border-top: 1px solid #dfe2e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99;
  -webkit-transition: -webkit-transform 0.2s ease-out;
  -webkit-transition: transform 0.2s ease-out;
  transition: transform 0.2s ease-out;
}

@media (max-width:450px) {
  .emotion-8 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    text-align: center;
    padding: 16px 0;
  }
}

.emotion-0 {
  margin-left: 16px;
  margin-right: 16px;
}

@media (max-width:450px) {
  .emotion-0:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-6 {
  margin-left: 16px;
  margin-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:450px) {
  .emotion-6:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
  width: 100%;
  max-width: 100%;
}

.emotion-4:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-4:disabled,
.emotion-4:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-4 span .wd-icon-fill,
.emotion-4 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #ffffff;
}

.emotion-4:focus {
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:focus span .wd-icon-fill,
.emotion-4:focus span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:hover:focus {
  background-color: #005cb9;
}

.emotion-4:active,
.emotion-4:focus:active,
.emotion-4:hover:active {
  background-color: #004387;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:active span .wd-icon-fill,
.emotion-4:focus:active span .wd-icon-fill,
.emotion-4:hover:active span .wd-icon-fill,
.emotion-4:active span .wd-icon-accent,
.emotion-4:focus:active span .wd-icon-accent,
.emotion-4:hover:active span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:hover {
  background-color: #005cb9;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:hover span .wd-icon-fill,
.emotion-4:hover span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:disabled,
.emotion-4:active:disabled,
.emotion-4:focus:disabled,
.emotion-4:hover:disabled {
  background-color: #a6d2ff;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:disabled span .wd-icon-fill,
.emotion-4:active:disabled span .wd-icon-fill,
.emotion-4:focus:disabled span .wd-icon-fill,
.emotion-4:hover:disabled span .wd-icon-fill,
.emotion-4:disabled span .wd-icon-accent,
.emotion-4:active:disabled span .wd-icon-accent,
.emotion-4:focus:disabled span .wd-icon-accent,
.emotion-4:hover:disabled span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-4:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-2 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-2:hover:active {
  background-color: transparent;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-0 emotion-1"
  >
    We use cookies to ensure that we give you the best experience on our website. If you continue without changing your settings, weâ€™ll assume that you are willing to receive cookies.
  </div>
  <div
    className="emotion-6 emotion-1"
  >
    <button
      className="emotion-4 emotion-5"
      onClick={[Function]}
    >
      <span
        className="emotion-2 emotion-3"
      >
        Continue
      </span>
    </button>
  </div>
</div>
`;

exports[`Cookie Banner Snapshots renders a CookieBanner with settings 1`] = `
.emotion-10 {
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffffff;
  border-top: 1px solid #dfe2e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99;
  -webkit-transition: -webkit-transform 0.2s ease-out;
  -webkit-transition: transform 0.2s ease-out;
  transition: transform 0.2s ease-out;
}

@media (max-width:450px) {
  .emotion-10 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    text-align: center;
    padding: 16px 0;
  }
}

.emotion-0 {
  margin-left: 16px;
  margin-right: 16px;
}

@media (max-width:450px) {
  .emotion-0:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-8 {
  margin-left: 16px;
  margin-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:450px) {
  .emotion-8:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
  width: 100%;
  max-width: 100%;
}

.emotion-6:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-6:disabled,
.emotion-6:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-6 span .wd-icon-fill,
.emotion-6 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #ffffff;
}

.emotion-6:focus {
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:focus span .wd-icon-fill,
.emotion-6:focus span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:hover:focus {
  background-color: #005cb9;
}

.emotion-6:active,
.emotion-6:focus:active,
.emotion-6:hover:active {
  background-color: #004387;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:active span .wd-icon-fill,
.emotion-6:focus:active span .wd-icon-fill,
.emotion-6:hover:active span .wd-icon-fill,
.emotion-6:active span .wd-icon-accent,
.emotion-6:focus:active span .wd-icon-accent,
.emotion-6:hover:active span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:hover {
  background-color: #005cb9;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:hover span .wd-icon-fill,
.emotion-6:hover span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:disabled,
.emotion-6:active:disabled,
.emotion-6:focus:disabled,
.emotion-6:hover:disabled {
  background-color: #a6d2ff;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:disabled span .wd-icon-fill,
.emotion-6:active:disabled span .wd-icon-fill,
.emotion-6:focus:disabled span .wd-icon-fill,
.emotion-6:hover:disabled span .wd-icon-fill,
.emotion-6:disabled span .wd-icon-accent,
.emotion-6:active:disabled span .wd-icon-accent,
.emotion-6:focus:disabled span .wd-icon-accent,
.emotion-6:hover:disabled span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-4 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-4:hover:active {
  background-color: transparent;
}

.emotion-2 {
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 16px;
  border: 0;
  font-weight: 500;
  white-space: nowrap;
  cursor: pointer;
  padding: 0;
  height: 0%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

<div
  className="emotion-10 emotion-11"
>
  <div
    className="emotion-0 emotion-1"
  >
    We use cookies to ensure that we give you the best experience on our website. If you continue without changing your settings, weâ€™ll assume that you are willing to receive cookies.
  </div>
  <div
    className="emotion-8 emotion-1"
  >
    <button
      className="emotion-2 emotion-3"
      onClick={[Function]}
    >
      Cookie Settings
    </button>
    <button
      className="emotion-6 emotion-7"
      onClick={[Function]}
    >
      <span
        className="emotion-4 emotion-5"
      >
        Continue
      </span>
    </button>
  </div>
</div>
`;

exports[`Cookie Banner Snapshots renders a closed CookieBanner 1`] = `
.emotion-0 {
  margin-left: 16px;
  margin-right: 16px;
}

@media (max-width:450px) {
  .emotion-0:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-6 {
  margin-left: 16px;
  margin-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:450px) {
  .emotion-6:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
  width: 100%;
  max-width: 100%;
}

.emotion-4:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-4:disabled,
.emotion-4:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-4 span .wd-icon-fill,
.emotion-4 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #ffffff;
}

.emotion-4:focus {
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:focus span .wd-icon-fill,
.emotion-4:focus span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:hover:focus {
  background-color: #005cb9;
}

.emotion-4:active,
.emotion-4:focus:active,
.emotion-4:hover:active {
  background-color: #004387;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:active span .wd-icon-fill,
.emotion-4:focus:active span .wd-icon-fill,
.emotion-4:hover:active span .wd-icon-fill,
.emotion-4:active span .wd-icon-accent,
.emotion-4:focus:active span .wd-icon-accent,
.emotion-4:hover:active span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:hover {
  background-color: #005cb9;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:hover span .wd-icon-fill,
.emotion-4:hover span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:disabled,
.emotion-4:active:disabled,
.emotion-4:focus:disabled,
.emotion-4:hover:disabled {
  background-color: #a6d2ff;
  border-color: transparent;
  color: #ffffff;
}

.emotion-4:disabled span .wd-icon-fill,
.emotion-4:active:disabled span .wd-icon-fill,
.emotion-4:focus:disabled span .wd-icon-fill,
.emotion-4:hover:disabled span .wd-icon-fill,
.emotion-4:disabled span .wd-icon-accent,
.emotion-4:active:disabled span .wd-icon-accent,
.emotion-4:focus:disabled span .wd-icon-accent,
.emotion-4:hover:disabled span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-4:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-4:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-2 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-2:hover:active {
  background-color: transparent;
}

.emotion-8 {
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffffff;
  border-top: 1px solid #dfe2e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99;
  -webkit-transition: -webkit-transform 0.2s ease-out;
  -webkit-transition: transform 0.2s ease-out;
  transition: transform 0.2s ease-out;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

@media (max-width:450px) {
  .emotion-8 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    text-align: center;
    padding: 16px 0;
  }
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-0 emotion-1"
  >
    We use cookies to ensure that we give you the best experience on our website. If you continue without changing your settings, weâ€™ll assume that you are willing to receive cookies.
  </div>
  <div
    className="emotion-6 emotion-1"
  >
    <button
      className="emotion-4 emotion-5"
      onClick={[Function]}
    >
      <span
        className="emotion-2 emotion-3"
      >
        Continue
      </span>
    </button>
  </div>
</div>
`;

exports[`Cookie Banner Snapshots renders a closed CookieBanner with settings 1`] = `
.emotion-0 {
  margin-left: 16px;
  margin-right: 16px;
}

@media (max-width:450px) {
  .emotion-0:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-8 {
  margin-left: 16px;
  margin-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:450px) {
  .emotion-8:not(:first-child) {
    margin-top: 16px;
  }
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
  width: 100%;
  max-width: 100%;
}

.emotion-6:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-6:disabled,
.emotion-6:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-6 span .wd-icon-fill,
.emotion-6 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #ffffff;
}

.emotion-6:focus {
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:focus span .wd-icon-fill,
.emotion-6:focus span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:hover:focus {
  background-color: #005cb9;
}

.emotion-6:active,
.emotion-6:focus:active,
.emotion-6:hover:active {
  background-color: #004387;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:active span .wd-icon-fill,
.emotion-6:focus:active span .wd-icon-fill,
.emotion-6:hover:active span .wd-icon-fill,
.emotion-6:active span .wd-icon-accent,
.emotion-6:focus:active span .wd-icon-accent,
.emotion-6:hover:active span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:hover {
  background-color: #005cb9;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:hover span .wd-icon-fill,
.emotion-6:hover span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:disabled,
.emotion-6:active:disabled,
.emotion-6:focus:disabled,
.emotion-6:hover:disabled {
  background-color: #a6d2ff;
  border-color: transparent;
  color: #ffffff;
}

.emotion-6:disabled span .wd-icon-fill,
.emotion-6:active:disabled span .wd-icon-fill,
.emotion-6:focus:disabled span .wd-icon-fill,
.emotion-6:hover:disabled span .wd-icon-fill,
.emotion-6:disabled span .wd-icon-accent,
.emotion-6:active:disabled span .wd-icon-accent,
.emotion-6:focus:disabled span .wd-icon-accent,
.emotion-6:hover:disabled span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-6:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-4 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-4:hover:active {
  background-color: transparent;
}

.emotion-10 {
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffffff;
  border-top: 1px solid #dfe2e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99;
  -webkit-transition: -webkit-transform 0.2s ease-out;
  -webkit-transition: transform 0.2s ease-out;
  transition: transform 0.2s ease-out;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

@media (max-width:450px) {
  .emotion-10 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    text-align: center;
    padding: 16px 0;
  }
}

.emotion-2 {
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 16px;
  border: 0;
  font-weight: 500;
  white-space: nowrap;
  cursor: pointer;
  padding: 0;
  height: 0%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

<div
  className="emotion-10 emotion-11"
>
  <div
    className="emotion-0 emotion-1"
  >
    We use cookies to ensure that we give you the best experience on our website. If you continue without changing your settings, weâ€™ll assume that you are willing to receive cookies.
  </div>
  <div
    className="emotion-8 emotion-1"
  >
    <button
      className="emotion-2 emotion-3"
      onClick={[Function]}
    >
      Cookie Settings
    </button>
    <button
      className="emotion-6 emotion-7"
      onClick={[Function]}
    >
      <span
        className="emotion-4 emotion-5"
      >
        Continue
      </span>
    </button>
  </div>
</div>
`;
